Automattic/jetpack
# Premium Content block: Fix 404 on login for posts with encoded URLs

Fixes [#603](603-gh-Automattic/dotcom-manage)

#### Changes proposed in this Pull Request:

* We now properly encode the post URL. Before we were encoding URLs that had the params already encoded, now we decode the URL before we encode the whole URL.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Env: Simple and/or Atomic
* Create a new Post
* Add a title with special chars to the Post, for example, 平常心面對拋物式下跌
* Add a premium content block to the Post and publish the Post.
* Open the Post in the Front End (make sure you aren't admin) and click on the login button, log in with a non-admin user.
* You should be redirected to the current Post instead of being redirected to a 404 page.

#### Testing evidence

https://user-images.githubusercontent.com/1989914/156001051-a0858baf-d1d1-442f-a366-0f6e5e0332ea.mp4
